### PR TITLE
[v1.x] Onnx export support for gather_nd

### PR DESCRIPTION
**Pull request by waytrue17**
## Description ##
Onnx export support for gather_nd

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: tests/python-pytest/onnx/test_operators.py [hunk 1]
```diff
@@ -452,3 +452,15 @@ def test_onnx_export_reshape_like(tmp_path, dtype):
     op_export_test('reshape_like3', M3, [x, y], tmp_path)
     M4 = def_model('reshape_like', lhs_begin=0, lhs_end=None, rhs_begin=1, rhs_end=None)
     op_export_test('reshape_like4', M4, [x, y], tmp_path)
+
+
+@pytest.mark.parametrize('dtype', ['int32', 'int64', 'float16', 'float32', 'float64'])
+def test_onnx_export_gather_nd(tmp_path, dtype):
+    x1 = mx.nd.array([[[1, 2], [3, 4]], [[5, 6], [7, 8]]], dtype=dtype)
```
Comment (Zha0q1) [inline tests/python-pytest/onnx/test_operators.py:459]: Shall we use high dimensional inputs and randomized inputs? something like
```
x = mx.random.uniform(0, 100, (7, 7, 7)).astype(dtype)
y = mx.random.uniform(0, 7, shape ).astype(dtype)
```

**Review by josephevans — APPROVED** [review]
LGTM!

File: tests/python-pytest/onnx/test_operators.py (hunk 1 above)
Comment (waytrue17) [reply tests/python-pytest/onnx/test_operators.py]: Fixed

**Review by Zha0q1 — APPROVED** [review]
lgtm thanks!

**Pull request merged**